server/service/src/requisition/program_settings/map.rs

```diff
@@ -0,0 +1,93 @@
+use repository::{
+    PeriodRow, ProgramRequisitionOrderTypeRow, ProgramRequisitionSettings, RequisitionsInPeriod,
+};
+
+use super::{prepare::PrepareProgramSettings, OrderType, ProgramSettings};
+
+/// Map prepared program_settings, order_types, periods and requisitions_in_periods to ProgramSettings
+/// order types are mapped to program settings by program setting id
+/// available periods are mapped with period_is_available, see description
+/// supplier are mapped by program_id
+pub(super) fn map(
+    PrepareProgramSettings {
+        settings,
+        order_types,
+        periods,
+        program_suppliers,
+        requisitions_in_period,
+    }: PrepareProgramSettings,
+) -> Vec<ProgramSettings> {
+    settings
+        .into_iter()
+        .map(|program_setting| {
+            // Filter and map (to include available_periods) order type for program setting
+            let order_types = order_types
+                .iter()
+                .filter(|o| {
+                    o.program_requisition_settings_id == program_setting.program_settings_row.id
+                })
+                .map(|order_type| {
+                    // Filter periods for order type, period_is_available
+                    let available_periods = periods
```

Comment:
I'm fine with this filtering, I wonder if it would be best from a UX perspective if we returned the periods and a status.
Then the UI could show that the period has reached it's max?
Probably not worth it at this stage, but it came to mind.

Reply:
I think periods would ideally have `counts` and `max`, all the info is there now really